### PR TITLE
update vmbackup/vmrestore README usage

### PR DESCRIPTION
**Pull request by f41gh7**
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/381

with minio and configuration file examples.

### CODE REVIEW

File: app/vmbackup/README.md [hunk 1]
```diff
@@ -9,9 +9,9 @@ Supported storage systems for backups:
 * Any S3-compatible storage such as [MinIO](https://github.com/minio/minio), [Ceph](https://docs.ceph.com/docs/mimic/radosgw/s3/) or [Swift](https://www.swiftstack.com/docs/admin/middleware/s3_middleware.html). See `-customS3Endpoint` command-line flag.
 * Local filesystem. Example: `fs://</absolute/path/to/backup>`
 
-Incremental backups and full backups are supported. Incremental backups are created automatically if the destination path already contains data from the previous backup.
+Incremental backups and full backups supported. Incremental backups created automatically if the destination path already contains data from the previous backup.
```
Comment (hagen1778) [inline app/vmbackup/README.md:12]: It was correct in both cases before. But if you don't like how it sounds - feel free to re-phrase. Like "vmbackup supports incremental and full backups."

File: app/vmbackup/README.md [hunk 2]
```diff
@@ -100,27 +100,27 @@ The backup algorithm is the following:
 2. Determine files in `-dst`, which are missing in `-snapshotName`, and delete them. These are usually small files, which are already merged into bigger files in the snapshot.
 3. Determine files from `-snapshotName`, which are missing in `-dst`. These are usually small new files and bigger merged files.
 4. Determine files from step 3, which exist in the `-origin`, and perform server-side copy of these files from `-origin` to `-dst`.
-   This are usually the biggest and the oldest files, which are shared between backups.
-5. Upload the remaining files from setp 3 from `-snapshotName` to `-dst`.
+   It is usually the biggest, and the oldest files, which are shared between backups.
```
Comment (hagen1778) [inline app/vmbackup/README.md:103]: "These are"
And no commas needed at all

File: app/vmbackup/README.md [hunk 3]
```diff
@@ -100,27 +100,27 @@ The backup algorithm is the following:
...
 - All the files in the snapshot are immutable.
-- Old files are periodically merged into new files.
+- Old files periodically merged into new files.
 - Smaller files have higher probability to be merged.
 - Consecutive snapshots share many identical files.
 
 These properties allow performing fast and cheap incremental backups and server-side copying from `-origin` paths.
 See [this article](https://medium.com/@valyala/speeding-up-backups-for-big-time-series-databases-533c1a927883) for more details.
-`vmbackup` can work improperly or slowly when these properties are violated.
+`vmbackup` can work improperly or slowly when these properties violated.
```
Comment (hagen1778) [inline app/vmbackup/README.md:118]: it sounded better before

File: app/vmbackup/README.md [hunk 4]
```diff
@@ -129,7 +129,46 @@ See [this article](https://medium.com/@valyala/speeding-up-backups-for-big-time-
 
 ### Advanced usage
 
-Run `vmbackup -help` in order to see all the available options:
+
+* Obtaining credentials from a file.
+  
+  Add flag `-credsFilePath=/etc/credentials` with following content:
```
Comment (hagen1778) [inline app/vmbackup/README.md:135]:
```suggestion
  Add flag `-credsFilePath=/etc/credentials` with the following content:
```

File: app/vmbackup/README.md [hunk 5]
```diff
@@ -129,7 +129,46 @@ See [this article](https://medium.com/@valyala/speeding-up-backups-for-big-time-
...
+           "private_key": "-----BEGIN PRIVATE KEY-----\nprivate-key\n-----END PRIVATE KEY-----\n",
+           "client_email": "service-account-email",
+           "client_id": "client-id",
+           "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+           "token_uri": "https://accounts.google.com/o/oauth2/token",
+           "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+           "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/service-account-email"
+         }
+    ```
+* Usage with s3 custom url endpoint.  It is possible to use `vmbackup` with s3 api compatible storages, like  minio, cloudian and other.
```
Comment (hagen1778) [inline app/vmbackup/README.md:160]:
```suggestion
* Usage with s3 custom url endpoint. It is possible to use `vmbackup` with s3 compatible storages like minio, cloudian, etc. 
```

File: app/vmbackup/README.md [hunk 6]
```diff
@@ -129,7 +129,46 @@ See [this article](https://medium.com/@valyala/speeding-up-backups-for-big-time-
...
+           "client_email": "service-account-email",
+           "client_id": "client-id",
+           "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+           "token_uri": "https://accounts.google.com/o/oauth2/token",
+           "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+           "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/service-account-email"
+         }
+    ```
+* Usage with s3 custom url endpoint.  It is possible to use `vmbackup` with s3 api compatible storages, like  minio, cloudian and other. 
+  You have to add custom url endpoint with a flag:
```
Comment (hagen1778) [inline app/vmbackup/README.md:161]:
```suggestion
  You have to add a custom url endpoint via flag: 
```

File: app/vmbackup/README.md [hunk 7]
```diff
@@ -129,7 +129,46 @@ See [this article](https://medium.com/@valyala/speeding-up-backups-for-big-time-
...
+    ```
+* Usage with s3 custom url endpoint.  It is possible to use `vmbackup` with s3 api compatible storages, like  minio, cloudian and other. 
+  You have to add custom url endpoint with a flag: 
+```
+  # for minio:  
+  -customS3Endpoint=http://localhost:9000
+
+  # for aws gov region 
+  -customS3Endpoint=https://s3-fips.us-gov-west-1.amazonaws.com
+
```
Comment (hagen1778) [inline app/vmbackup/README.md:168]:
```suggestion
```

File: app/vmbackup/README.md [hunk 8]
```diff
@@ -205,9 +244,9 @@ It is recommended using [binary releases](https://github.com/VictoriaMetrics/Vic
 
 Run `make package-vmbackup`. It builds `victoriametrics/vmbackup:<PKG_TAG>` docker image locally.
 `<PKG_TAG>` is auto-generated image tag, which depends on source code in the repository.
-The `<PKG_TAG>` may be manually set via `PKG_TAG=foobar make package-vmbackup`.
+The `<PKG_TAG>` maybe manually set via `PKG_TAG=foobar make package-vmbackup`.
```
Comment (hagen1778) [inline app/vmbackup/README.md:247]: it was correct before

File: app/vmbackup/README.md [hunk 9]
```diff
@@ -205,9 +244,9 @@ It is recommended using [binary releases](https://github.com/VictoriaMetrics/Vic
...
 
-By default the image is built on top of [alpine](https://hub.docker.com/_/alpine) image. It is possible to build the package on top of any other base image
+By default, the image builds on top of [alpine](https://hub.docker.com/_/alpine) image. It is possible to build the package on top of any other base image
```
Comment (hagen1778) [inline app/vmbackup/README.md:249]:
```suggestion
The base docker image is [alpine](https://hub.docker.com/_/alpine) but it is possible to use any other base image
```

File: app/vmrestore/README.md [hunk 10]
```diff
@@ -9,31 +9,69 @@ when restarting `vmrestore` with the same args.
 
 ### Usage
 
-VictoriaMetrics must be stopped during the restore process.
+VictoriaMetrics must be stopped during restore process.
```
Comment (hagen1778) [inline app/vmrestore/README.md:12]: it was correct before

File: app/vmrestore/README.md [hunk 11]
```diff
@@ -9,31 +9,69 @@ when restarting `vmrestore` with the same args.
...
 
 ```
 vmrestore -src=gcs://<bucket>/<path/to/backup> -storageDataPath=<local/path/to/restore>
 
 ```
 
 * `<bucket>` is [GCS bucket](https://cloud.google.com/storage/docs/creating-buckets) name.
-* `<path/to/backup>` is the path to backup made with [vmbackup](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vmbackup/README.md) on GCS bucket.
+* `<path/to/backup>` is the path to back up made with [vmbackup](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vmbackup/README.md) on GCS bucket.
```
Comment (hagen1778) [inline app/vmrestore/README.md:20]: it was correct before - backup is noun here

File: app/vmrestore/README.md [hunk 12]
```diff
@@ -9,31 +9,69 @@ when restarting `vmrestore` with the same args.
...
 * `<local/path/to/restore>` is the path to folder where data will be restored. This folder must be passed
-  to VictoriaMetrics in `-storageDataPath` command-line flag after the restore process is complete.
+  to VictoriaMetrics in `-storageDataPath` command-line flag after restore process is complete.
```
Comment (hagen1778) [inline app/vmrestore/README.md:22]: it was correct before

File: app/vmrestore/README.md [hunk 13]
```diff
@@ -9,31 +9,69 @@ when restarting `vmrestore` with the same args.
...
-The original `-storageDataPath` directory may contain old files. They will be susbstituted by the files from backup,
+The original `-storageDataPath` directory may contain old files. They will be substituted by the files from backup,
 i.e. the end result would be similar to [rsync --delete](https://askubuntu.com/questions/476041/how-do-i-make-rsync-delete-files-that-have-been-deleted-from-the-source-folder).
 
 
 ### Troubleshooting
 
 * If `vmrestore` eats all the network bandwidth, then set `-maxBytesPerSecond` to the desired value.
-* If `vmrestore` has been interrupted due to temporary error, then just restart it with the same args. It will resume the restore process.
+* If `vmrestore` has been interrupted due to temporary error, then just restart it with the same args. It will resume restore process.
```
Comment (hagen1778) [inline app/vmrestore/README.md:31]: it was correct before

**Review by hagen1778 — APPROVED** [review]
lgtm except some comments

File: app/vmrestore/README.md [hunk 14]
```diff
@@ -102,9 +140,9 @@ It is recommended using [binary releases](https://github.com/VictoriaMetrics/Vic
 
 Run `make package-vmrestore`. It builds `victoriametrics/vmrestore:<PKG_TAG>` docker image locally.
 `<PKG_TAG>` is auto-generated image tag, which depends on source code in the repository.
-The `<PKG_TAG>` may be manually set via `PKG_TAG=foobar make package-vmrestore`.
+The `<PKG_TAG>` maybe manually set via `PKG_TAG=foobar make package-vmrestore`.
```
Comment (valyala) [inline app/vmrestore/README.md:143]:
```suggestion
The `<PKG_TAG>` may be manually set via `PKG_TAG=foobar make package-vmrestore`.
```

File: app/vmrestore/README.md [hunk 15]
```diff
@@ -102,9 +140,9 @@ It is recommended using [binary releases](https://github.com/VictoriaMetrics/Vic
...
 
-By default the image is built on top of [alpine](https://hub.docker.com/_/alpine) image. It is possible to build the package on top of any other base image
+By default, the image builds on top of [alpine](https://hub.docker.com/_/alpine) image. It is possible to build the package on top of any other base image
```
Comment (valyala) [inline app/vmrestore/README.md:145]:
```suggestion
The base docker image is [alpine](https://hub.docker.com/_/alpine) but it is possible to use any other base image
```

File: README.md [hunk 16]
```diff
@@ -896,6 +896,14 @@ VictoriaMetrics also may scrape Prometheus targets - see [these docs](#how-to-sc
 
 VictoriaMetrics supports Prometheus-compatible relabeling for all the ingested metrics if `-relabelConfig` command-line flag points
 to a file containing a list of [relabel_config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config) entries.
```
Comment (valyala) [inline README.md:898]:
```suggestion
to a file containing a list of [relabel_config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config) entries.
Example contents for `-relabelConfig` file:
```

**Review by valyala — APPROVED** [review]

**Review by valyala — APPROVED** [review]

**Pull request merged**